runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ActionListPanel/ActionListPanel.tsx

```diff
@@ -24,34 +24,45 @@ import ListGroup, { Item } from '../ListGroup';
 import internalBpmnActions from '../ConfigureActionPanel/Actions';
 import { ActionToBPMNElement, TaskType } from '../ConfigureActionPanel/ActionToBPMNElement';
 import { IBpmnAction, Runner } from '../ConfigureActionPanel/Actions/types';
-import { defaultActionGroups, defaultTemplatesGroups } from '../ConfigureActionPanel/DefaultGroups';
+import { useTemplatesGroups } from '../ConfigureActionPanel/useTemplatesGroups';
+import useInternalActionsGroups from '../ConfigureActionPanel/useInternalActionsGroups';
 import { internalTemplates } from '../ConfigureActionPanel/Templates';
 import CustomTemplateHandler from '../ConfigureActionPanel/CustomTemplateHandler';
 import customLoopHandler from '../ConfigureActionPanel/CustomLoopHandler';
 import FilterModal from './FilterModal';
+import i18n from 'src/translations/i18n';
 
 const filterModalInitialState: FilterModalState = {
     anchorElement: null,
     filters: [],
 };
 
 const ActionListPanel: FC<ActionListPanelProps> = memo((props) => {
-    const [actionGroups, setActionGroups] = useState(defaultActionGroups);
+    const internalActionsGroups = useInternalActionsGroups();
+    const templatesGroups = useTemplatesGroups()
     const [currentTab, setCurrentTab] = useState<ListPanelTab>(ListPanelTab.ACTIONS);
     const [filterModal, setFilterModal] = useState<FilterModalState>(filterModalInitialState);
     const filterModalAnchorRef = useRef<Element>(null);
-    const templateGroupsList = Object.entries(defaultTemplatesGroups);
+    const templatesGroupsList = Object.entries(templatesGroups);
     const { byId, external } = useSelector((state) => state.action.bpmnActions);
-
+    
     const [open, setOpen] = useState(true);
     const { translate } = useTranslations();
-    const actionGroupsList = React.useMemo(() => Object.entries(actionGroups), [actionGroups]);
+
+    const actionsGroups = React.useMemo(() => {
+        const externalActionsGroup = external.map((externalId) => byId[externalId]);
```

Comment:
that's not a fully functional programming approach. Try:

`const completeActionsGroups = { ...internalActionsGroups, external: { ...internalActionsGroups.external, items: [...externalActionsGroup]} };`

or use `immer` package, see https://immerjs.github.io/immer/example-setstate

Reply:
did it before see it!